packages/js-runtime/src/runtime/URL.ts

```diff
@@ -143,6 +152,9 @@ export class URL {
 
       if (['http:', 'https:'].includes(protocol) || ['blob:', 'file:'].includes(origin)) {
         this.origin = protocol + '//' + hostname;
+        if (port && port !== DEFAULT_PORTS[protocol]) {
+          this.origin += ':' + port;
```

Comment:
Shouldn't we also update `this.port` according to `DEFAULT_PORTS`? Reading the spec, it looks like we should also set a default port based on the scheme, if no custom port was defined.

Reply:
I couldn't grok the wording but none of Chrome, Firefox, Node, or Deno does it. Actually I just noticed they all return an empty string for  `new URL("https://google.com:443").port` and `new URL("http://google.com:80").port`. So I guess we're supposed to do the opposite and empty the `port` property if it's a default port?

Reply:
2.1.3 under https://url.spec.whatwg.org/#port-state says:

> Set url’s [port](https://url.spec.whatwg.org/#concept-url-port) to null, if port is url’s [scheme](https://url.spec.whatwg.org/#concept-url-scheme)’s [default port](https://url.spec.whatwg.org/#default-port); otherwise to port.

So we're supposed to _remove_ the default port, not add it, like all the other implementations. Will do at the first opportunity.

Reply:
Just pushed the version that returns empty string for the default port.